sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -627,20 +627,20 @@ private void queueWork(int maximumMessageCount, Duration maxWaitTime,
         FluxSink<ServiceBusReceivedMessageContext> emitter) {
         synchronized (lock) {
             final long id = idGenerator.getAndIncrement();
-            EmitterProcessor<ServiceBusReceivedMessageContext> emitterProcessor = messageProcessor.get();
-
             final SynchronousReceiveWork work = new SynchronousReceiveWork(id, maximumMessageCount, maxWaitTime,
                 emitter);
-            final SynchronousMessageSubscriber syncSubscriber = new SynchronousMessageSubscriber(work);
-            logger.info("[{}]: Started synchronous message subscriber.", id);
 
-            if (emitterProcessor == null) {
-                emitterProcessor = this.asyncClient.receive(DEFAULT_RECEIVE_OPTIONS)
-                    .subscribeWith(EmitterProcessor.create(asyncClient.getReceiverOptions().getPrefetchCount(), false));
-                messageProcessor.set(emitterProcessor);
+            SynchronousMessageSubscriber messageSubscriber = synchronousMessageSubscriber.get();
+            if (messageSubscriber == null) {
+                messageSubscriber = asyncClient.receive(DEFAULT_RECEIVE_OPTIONS)
+                    .subscribeWith(new SynchronousMessageSubscriber(asyncClient.getReceiverOptions()
+                        .getPrefetchCount(), work));
+                synchronousMessageSubscriber.set(messageSubscriber);
```

Comment:
This isn't thread-safe.
It's possible two different threads see `messageSubscriber == null`, and create the same instance of Subscriber. Then the last one wins and the first one is lost.